### DISCUSSION TITLE
Accessing stats["_trades"] during the test

### DISCUSSION THREAD

**Discussion by MaxvanderWerf** [kernc/backtesting.py#722] [Q&A] [answered]
I'm trying to code a variation of Dollar Cost Averaging and therefore need to know the trades that have already been made during the test to determine if I should buy or not. Initializing self.stats = self.stats["_trades"] in the strategy gives me an error. Does anyone know if this is possible at all?
```py
class DCA_every_week(Strategy):

    amount_to_invest = 10000

    def init(self):
        
        self.day_of_week = self.I(
                            lambda x: x,
                            self.data.Close.s.index.dayofweek,
                            plot = False,
                            )
        
        self.stats = self.stats["_trades"]

    def next(self):

        if self.day_of_week[-1] == 4: 
            self.buy(size = math.floor(self.amount_to_invest / self.data.Close[-1])) # Buys largest number of stocks possible
```
```
AttributeError: 'DCA_every_week' object has no attribute 'stats'
```

**Comment by kernc** [comment, marked as answer]
There is no `Strategy.stats`! `stats` are _returned_ by [`Backtest.run()`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Backtest.run).
During the run, you can maybe look in [`Strategy.closed_trades`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.closed_trades)?

**Comment by MaxvanderWerf** [comment]
Thanks! If anyone else is wondering, I found them using Strategy.trades (with DCA you only buy and don't sell).

**Comment by MaxvanderWerf** [comment]
One more question:

I can get the length of the list of active trades through len(self.trades), however I cannot access the variables of trades.

When I try:
self.trades.entry_time
I get:
AttributeError: 'tuple' object has no attribute 'entry_time'

When I try to get the latest trade by using:
self.trades[0].entry_time
I get:
IndexError: tuple index out of range

Does anyone know how to access the instance variables of a trade?

**Reply by kernc** [reply]
> When I try to get the latest trade by using: self.trades[0].entry_time I get: IndexError: tuple index out of range

At the beginning when there are no trades yet, `0` in `trades[0]` is an invalid index. Hide it behind `if self.trades:` check.
